File: swift/Bob.swift
import Foundation

class Bob {
    class func hey(_ s: String) -> String {
        if s.uppercased() == s && s.rangeOfCharacter(from: CharacterSet.letters) != nil {
            return "Whoa, chill out!"
        }
        if (s.filter { $0 != " " }).isEmpty { return "Fine. Be that way!" }
        if s.suffix(1) == "?" { return "Sure." }

        return "Whatever."
    }
}

File: clojure/bob.clj
(ns bob
  (:require [clojure.string :as string]))

(defn response-for
  "Determines, and returns Bob's response in a conversation."
  [s]
  (let [fs (string/replace s #"[^a-zA-Z]" "")]
    (cond 
      (and (= s (string/upper-case s))
           (not (string/blank? fs))) "Whoa, chill out!"
      (string/blank? s) "Fine. Be that way!"
      (string/ends-with? s "?") "Sure."
      :else "Whatever.")))
